### ISSUE TITLE
The difference of RFBNet in RFB_Net_vgg.py and Fig.5 ?

### ISSUE THREAD

**Issue by LightToYang** [ruinmessi/RFBNet#27]
Hello, thank you for code releasing. 

There are two branches from fc7 layer output  in Fig.5, one is the input of RBF and another is the input of RBF(stride 2).

![v2-7d446ac4148e5cc73624808f2a7558ff_r 2](https://user-images.githubusercontent.com/17929130/44197533-7587a380-a171-11e8-84d8-aa0b768e39d9.jpg)

`       # apply vgg up to fc7
        for k in range(23, len(self.base)):
            x = self.base[k](x)

        # apply extra layers and cache source layer outputs
        for k, v in enumerate(self.extras):
            x = v(x)
            if k < self.indicator or k%2 ==0:
                sources.append(x)`

But, in RFB_Net_vgg.py, the output x from fc7 layer is the input of RFB layer, and then, the output of RFB layer is the input of RFB(stride 2) layer. There might not be two branches. And the architecture of RFBNet might be like the picture below?

![rfbnet-300-c](https://user-images.githubusercontent.com/17929130/44197717-0cecf680-a172-11e8-8ea4-d20f986c82cd.jpg)

**Comment by ruinmessi** [comment]
@LightToYang Yes, you are right! Thanks a lot for pointing out this mistake. And actually all my experiments are using the architecture like your picture. I just ignore this details when painting Fig.5, sorry for the confusion~

**Comment by zHanami** [comment]
Are you sure that RFB-s has applied to the conv4_3? I can not find the operations, there are only extra['300'] = [1024,512,256]，corresponding to RFB；What's more, in you paper Fig.4, the RFB consists of rate = [1,3,5], while the code is [2,3,5]

**Comment by ruinmessi** [comment]
@zHanami RFB-s is adding as self.Norm in RFB.

**Comment by zHanami** [comment]
@ruinmessi Thanks, I just mis-trust it as the L2norm in ssd，but can you figure out the rate in your paper and code, which I refer.

**Comment by ruinmessi** [comment]
@zHanami  Thanks, I check the code, you are right, the rate of the code is different from the paper.

**Comment by jiachen0212** [comment]
hello @LightToYang 
when i read the RFB_Netvgg.py, i also get some confused。  
that is the extra_layer:
i think the extras layer is: 
layers += [BasicRFB(1024, 1024, scale = 1.0, visual=2, stride=1)]
layers += [BasicRFB(1024, 512, scale = 1.0, visual=2, stride=2)]
layers += [BasicRFB(512, 256, scale = 1.0, visual=2 stride=2)]

layers += [BasicConv(256,128,kernel_size=1,stride=1)]
layers += [BasicConv(128,256,kernel_size=3,stride=1)]
layers += [BasicConv(256,128,kernel_size=1,stride=1)]
layers += [BasicConv(128,256,kernel_size=3,stride=1)]

so：
if k < self.indicator or k%2 ==0:
sources.append(x)：
k should be 0 1 2 4 6...... 🤔？   
may have some difference with your picture.....  can you help me？

**Comment by ZCCDL** [comment]
why it is 23? for k in range(23, len(self.base)):

**Comment by XDUSPONGE** [comment]
> why it is 23? for k in range(23, len(self.base)):

hello, you can have a detail look of the vgg, the 23 because we want to get the conv4_3 layer

**Comment by DonghunP** [comment]
> @zHanami Thanks, I check the code, you are right, the rate of the code is different from the paper.

It is deliberate or your mistake??

**Comment by DonghunP** [comment]
@LightToYang 
Did you solve your problem?
I agree with you.

The code should be changed like below or sth. 
```python
        # apply extra layers and cache source layer outputs
        for k, v in enumerate(self.extras):
            if k == 0:
                fc7 = x
            x = v(x)
            if k < self.indicator or k % 2 == 0:
                features.append(x)
            if k == 0:
                x = fc7
```

I didn't check the paper "same" code show better result.

**Comment by dongzhenguo2016** [comment]
` # apply extra layers and cache source layer outputs
        for k, v in enumerate(self.extras):
            if k == 0:
                fc7 = x
            x = v(x)
            if k < self.indicator or k % 2 == 0:
                features.append(x)
            if k == 0:
                x = fc7`

**Comment by dongzhenguo2016** [comment]
>  # apply extra layers and cache source layer outputs
        for k, v in enumerate(self.extras):
            if k == 0:
                fc7 = x
            x = v(x)
            if k < self.indicator or k % 2 == 0:
                features.append(x)
            if k == 0:
                x = fc7

**Comment by dongzhenguo2016** [comment]
> code
 # apply extra layers and cache source layer outputs
        for k, v in enumerate(self.extras):
            if k == 0:
                fc7 = x
            x = v(x)
            if k < self.indicator or k % 2 == 0:
                features.append(x)
            if k == 0:
                x = fc7
